coordinator/src/main/scala/filodb.coordinator/queryengine2/QueryEngine.scala

```diff
@@ -229,7 +229,13 @@ class QueryEngine(dataset: Dataset,
                                      options: QueryOptions,
                                      lp: SeriesKeysByFilters): Seq[PartKeysExec] = {
     val renamedFilters = renameMetricFilter(lp.filters)
-    shardsFromFilters(renamedFilters, options).map { shard =>
+    val filterCols = lp.filters.map(_.column).toSet
+    val shardsToHit = if (shardColumns.toSet.subsetOf(filterCols)) {
```

Comment:
This is fine although you can also just all shardsFromFilters with the shardOverrides option if the shardColumns are not part of the set.
Also we should probably track how many such queries there are.  Please add a stat.